cub/benchmarks/bench/topk/keys.cu

```diff
@@ -0,0 +1,136 @@
+// %RANGE% TUNE_BLOCK_LOAD_ALGORITHM ld 0:2:1
+
+#if TUNE_BLOCK_LOAD_ALGORITHM == 0
+#  define TUNE_LOAD_ALGORITHM cub::BLOCK_LOAD_DIRECT
+#elif TUNE_BLOCK_LOAD_ALGORITHM == 1
+#  define TUNE_LOAD_ALGORITHM cub::BLOCK_LOAD_WARP_TRANSPOSE
+#elif TUNE_BLOCK_LOAD_ALGORITHM == 2
+#  define TUNE_LOAD_ALGORITHM cub::BLOCK_LOAD_VECTORIZE
+#endif
+
+#if !TUNE_BASE
+template <class KeyInT, class OffsetT>
+struct policy_hub_t
+{
+  struct policy_t : cub::ChainedPolicy<300, policy_t, policy_t>
+  {
+    static constexpr int NOMINAL_4B_ITEMS_PER_THREAD = TUNE_ITEMS_PER_THREAD;
+    static constexpr int ITEMS_PER_THREAD = cuda::std::max(1, (NOMINAL_4B_ITEMS_PER_THREAD * 4 / sizeof(KeyInT)));
+
+    static constexpr int BITS_PER_PASS          = cub::detail::topk::calc_bits_per_pass<KeyInT>();
+    static constexpr int COEFFICIENT_FOR_BUFFER = 128;
+
+    using TopKPolicyT =
+      cub::AgentTopKPolicy<TUNE_THREADS_PER_BLOCK,
+                           ITEMS_PER_THREAD,
+                           BITS_PER_PASS,
+                           COEFFICIENT_FOR_BUFFER,
+                           TUNE_LOAD_ALGORITHM,
+                           cub::BLOCK_SCAN_WARP_SCANS>;
+  };
+
+  using MaxPolicy = policy_t;
+};
+#endif // !TUNE_BASE
+
+template <typename KeyT, typename OffsetT, typename OutOffsetT>
+void topk_keys(nvbench::state& state, nvbench::type_list<KeyT, OffsetT, OutOffsetT>)
+{
+  using key_input_it_t  = const KeyT*;
+  using key_output_it_t = KeyT*;
+  using offset_t        = cub::detail::choose_offset_t<OffsetT>;
+  using out_offset_t =
+    ::cuda::std::conditional_t<sizeof(offset_t) < sizeof(cub::detail::choose_offset_t<OutOffsetT>),
+                               offset_t,
+                               cub::detail::choose_offset_t<OutOffsetT>>;
+
+#if !TUNE_BASE
+  using policy_t   = policy_hub_t<KeyT, OffsetT>;
+  using dispatch_t = cub::detail::topk::DispatchTopK<
+    key_input_it_t,
+    key_output_it_t,
+    cub::NullType*,
+    cub::NullType*,
+    offset_t,
+    out_offset_t,
+    cub::detail::topk::select::max,
+    policy_t>;
+#else // TUNE_BASE
+  using dispatch_t = cub::detail::topk::DispatchTopK<
+    key_input_it_t,
+    key_output_it_t,
+    cub::NullType*,
+    cub::NullType*,
+    offset_t,
+    out_offset_t,
+    cub::detail::topk::select::max>;
+#endif // TUNE_BASE
+
+  // Retrieve axis parameters
+  const auto elements          = static_cast<size_t>(state.get_int64("Elements{io}"));
+  const auto selected_elements = static_cast<size_t>(state.get_int64("SelectedElements"));
+  const bit_entropy entropy    = str_to_entropy(state.get_string("Entropy"));
+
+  // Skip benchmarks at runtime
+  if (selected_elements >= elements)
+  {
+    state.skip("We only support the case where the variable K is smaller than the variable N.");
+    return;
+  }
+
+  thrust::device_vector<KeyT> in_keys  = generate(elements, entropy);
+  thrust::device_vector<KeyT> out_keys = generate(selected_elements);
```

Comment:
Q: Why do we need to generate random data for the output elements? Isn't this overwritten?

Similar question for `pairs.cu`